src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -105,6 +105,28 @@ object SnowflakeTelemetry {
     }
   }
 
+  private val MAX_CACHED_SPARK_PLAN_STATISTIC_COUNT = 1000
+
+  private[snowflake] def addSparkPlanStatistic(statistic: Set[String]): Unit = {
+    if (statistic.nonEmpty) {
+      this.synchronized {
+        if (logs.length >= MAX_CACHED_SPARK_PLAN_STATISTIC_COUNT) {
```

Comment:
If the pushdown is enabled, the spark plan statistic is added to `SnowflakeTelemetry.logs`, but it will be sent if any other kinds of messages is sent to snowflake lately.
In worst case, the pushdown is enabled, but the user doesn't not read/write from/to snowflake, all the spark plan statistic may be cached. To avoid OOM, SC cache at most 1000 messages.